commons/src/test/java/io/hyscale/commons/framework/patch/StrategicPatchTest.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.framework.patch;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.File;
+import java.io.StringReader;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import org.json.JSONException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import io.hyscale.commons.exception.CommonErrorCode;
+import io.hyscale.commons.exception.HyscaleErrorCode;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.utils.HyscaleFilesUtil;
+
+public class StrategicPatchTest {
+
+    private static String sourceData;
+    private static String patchData;
+    private static String mergedData;
+
+    @BeforeAll
+    public static void init() {
+        try {
+            sourceData = getData("/patch/source.json");
+            patchData = getData("/patch/patch.json");
+            mergedData = getData("/patch/merged.json");
+        } catch (HyscaleException e) {
+            fail();
```

Comment:
Any assertion failure should be a failure of unit test. Since it is a setup / init failure, throw a hyscale exception instead of failing the test. It gives wrong information to the end user.